rasterio/rio/features.py

```diff
@@ -122,14 +122,26 @@ def mask(
         bounds = geojson.get('bbox', calculate_bounds(geojson))
 
         with rasterio.open(input) as src:
-            has_disjoint_bounds = disjoint_bounds(bounds, src.bounds)
+            # If y pixel value is positive, then invert y dimension in bounds
+            invert_y = src.affine.e > 0
+
+            src_bounds = src.bounds
+            if invert_y:
+                src_bounds = [src.bounds[0], src.bounds[3],
+                              src.bounds[2], src.bounds[1]]
+
```

Comment:
Ah the dreaded upside-down raster. You must be working with NetCDFs too much :smile: 

I'm +0 on including this in the PR.  Great idea but I'd rather see this logic centralized into the base classes or a util function rather than implemented everywhere we needed to work with bounds. 

Handling inverted rasters (or a upside-down or whatever we call it) is a bigger issue that we need to look at across the entire code base. I did some ad hoc support in `rasterstats` and felt like something that would be better addressed in rasterio core. But I haven't take the time to do a full review of what that might take.